src/api/user/user.controller.ts

```diff
@@ -25,4 +25,26 @@ export class UserController {
       result: userData,
     };
   }
+
+  @Get(':id')
+  async getExample(@Param('id') id: string): Promise<any> {
+    const result = await this.exampleService.findExampleById(id);
+
+    // Simulate non-optimized code
+    const modifiedResult = this.processResult(result);
+
+    return modifiedResult;
+  }
+
+  // Non-optimized code
+  private processResult(result: any): any {
+    let modifiedResult = result;
+
+    // Example non-optimized code
+    for (let i = 0; i < modifiedResult.length; i++) {
```

Comment:
**Issue**: The loop used to convert names to uppercase is inefficient for large datasets. Each iteration modifies the array in-place, which can be costly in terms of performance. <br> **Fix**: Consider using a more functional approach with 'map', which can be more performant and easier to understand. Additionally, ensure that the operation is necessary before applying such transformations, as it could be handled at the database query level or through other means that do not require looping through the entire dataset in the application layer. <br> **Code Suggestion**: 
 ```
 - for (let i = 0; i < modifiedResult.length; i++) {
 -   modifiedResult[i].name = modifiedResult[i].name.toUpperCase();
 - }
 + modifiedResult = modifiedResult.map(item => ({
 +   ...item,
 +   name: item.name.toUpperCase()
 + }))
 ```